### PR TITLE
ICL-23917 - Adding before

### PR DESCRIPTION
**Pull request by svedbg**

### CODE REVIEW

File: bin/flok-show [hunk 1]
```diff
@@ -148,7 +148,7 @@ function rowOf(m, fields) {
 function allMigrations() {
   var m = flok.migrations.slice(0);
   m.sort(function (a, b) {
-    return a.status.runTime - b.status.runTime;
+    return a.status.runTime === b.status.runTime ? 0 : a.status.runTime < b.status.runTime ? -1 : 1;
```
Comment (seangarner) [inline bin/flok-show]: what was wrong with the subtraction method?

File: lib/flok.js [hunk 2]
```diff
@@ -303,6 +303,16 @@ Flok.prototype.sortMigrationsUp = function sortMigrationsUp(callback) {
...
     }
   }
+
+  const mBefore = filter(this.migrations, 'before');
+  for (let i = 0; i < mBefore.length; i++) {
+    const currentPos = this.migrations.indexOf(mBefore[i]);
+    const beforeDep = this.migrations.find(({id}) => id === mBefore[i].before);
+
+    const insertPos = this.migrations.indexOf(beforeDep);
+    console.dir(insertPos)
```
Comment (seangarner) [inline lib/flok.js]:
```suggestion
```

File: lib/flok.js [hunk 3]
```diff
@@ -303,6 +303,16 @@ Flok.prototype.sortMigrationsUp = function sortMigrationsUp(callback) {
       this.migrations.splice(pos, 0, this.migrations.splice(i, 1)[0]);
     }
   }
+
+  const mBefore = filter(this.migrations, 'before');
```
Comment (seangarner) [inline lib/flok.js]: The only drawback with resorting the migrations after dependencies are checked is that something could execute before it's dependencies.  Would a safer, and equally as simple, approach be to add the migration as a dep to the old migration referenced by the `before`?

* loop over migrations with a `before` property
* add the migration's id as a dependency of the migration referenced by `before`
* sort as normal

**Review by seangarner — CHANGES_REQUESTED** [review]

File: lib/flok.js (hunk 3 above)
Comment (svedbg) [reply lib/flok.js]: @seangarner is this what you mean: https://github.com/dynamicaction/flok/pull/6/commits/74c9a16e229321fab1ab3f4a84a8643fd73348c3?

File: bin/flok-show (hunk 1 above)
Comment (svedbg) [reply bin/flok-show]: I don't know I feel better when I'm using -1, 0 1 for sorting :)

Comment (seangarner) [reply bin/flok-show]: Well technically times will convert to numbers (ms) as you do any math on them, so it's the same thing.  But if it's easier to read them I'm ok with that.

File: lib/flok.js (hunk 3 above)
Comment (seangarner) [reply lib/flok.js]: Yep.  Perfect.

**Review by seangarner — APPROVED** [review]

**Pull request merged**